src/app/InteractionModelEngine.cpp

```diff
@@ -227,6 +227,58 @@ void InteractionModelEngine::OnResponseTimeout(Messaging::ExchangeContext * ec)
     ChipLogProgress(DataManagement, "Time out! failed to receive echo response from Exchange: %d", ec->GetExchangeId());
 }
 
+// Release clusterInfo list for the read handler to pool and shrink the rear of clusterInfo List forward if any
+void InteractionModelEngine::ReleaseClusterInfoListToPool(ReadHandler * const apReadHandler)
```

Comment:
why is this method not part of ReadHandler?

This handles internal buffer manipulation and resizing - I think the handler is more likely to correctly own these items

Reply:
yep,move it to readhandler, done, thanks